✨ #2156 Facet Panel Tabs

demariadaniel opened the pull request:
**Description of changes**

Implements Tabs for toggling between Clinical Filters and File Filters on the Facet Panel
Feature flagged

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [x] Modified Existing components and updates stories
  - [ ] No new UIKit components or changes
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

samrichca commented on components/pages/file-repository/FacetPanel/index.tsx:
```diff
@@ -448,25 +469,40 @@ const FacetPanel = () => {
+    padding: 5px 20px;
+    margin: 9px 5px 0px;
+    border: 1px solid ${theme.colors.grey_2};
+    border-top: ${currentTab === tabValue ? `4px solid ${theme.colors.secondary}` : null};
```
I would use `0` instead of `null` here. I don't think `null` will break anything per se, but it's not a standard CSS value.

demariadaniel replied:
good point, maybe i can use 'initial' or 'inherit', i was trying to 'short circuit' here so there are no changes unless it's the active tab

demariadaniel replied:
OK turns out the whole thing could be greatly simplified by just leveraging the `active` class, give it a 2nd look when you can. thanks!

samrichca replied:
even better. ok, ship it!

samrichca commented on components/pages/file-repository/FacetPanel/index.tsx:
```diff
@@ -448,25 +469,40 @@ const FacetPanel = () => {
+        {FEATURE_FACET_TABS_ENABLED && (
+          <FacetRow>
+            <Tabs value={currentTab} onChange={(e, value) => setTabs(value)}>
+              <Tab value="clinical" label="Clinical Filters" css={tabStyles('clinical')} />
```
the CSS prop here. nice!

demariadaniel replied:
thanks i wasn't sure how this would be received as the rest of the app is largely using css inline ;)

mistryrn reviewed (changes requested):
Looks good! We should probably also add this feature flag to the `.env.schema` as well. Once I added the flag to my .env it worked like a charm!

I'm not sure the exact process to add this flag to dev/qa/prod, I think it has something to do with helm charts. Someone like Henro would likely be able to point you in the right direction there.

demariadaniel commented:
> Looks good! We should probably also add this feature flag to the `.env.schema` as well. Once I added the flag to my .env it worked like a charm!
> 
> I'm not sure the exact process to add this flag to dev/qa/prod, I think it has something to do with helm charts. Someone like Henro would likely be able to point you in the right direction there.

Added to `env.schema`
PR for infra dev here: https://github.com/OICR/argo-infra/pull/694

mistryrn reviewed (approved):
🚀
